### DISCUSSION TITLE
OpenVPN AS w Azure

### DISCUSSION THREAD

**Discussion by TheRealMamuth** [AzureCommunityPL/group#162] [Pomoc] [answered]
Hej, Witam.
Mam problem po zestawieniu tunelu z instacją OpenVPN AS jak połacze sie po adresie maszyny wirtualnej 10.0.1.5 na przykład to po 30 sekundach mam drop połączenia SSH, nie wiem co to może być - ktoś miał podobny problem - już nie wiem gdzie sprawdzać co powoduje problemy. 

- siec domowa to 192.168.2.0/24, siec wirtualna w azure to 10.0.1.0/24
- IP forwarding na instancji z OpenVPN włączone w azure.
- Routing table ustawiony.
- Ja wejde po publicznym adresie IP tej vmki (10.0.1.0/24) sięgam normalnie moja siec domową
- testowane z samym routing i nat

**Comment by TheRealMamuth** [comment]
Problem rozwiązałem samodzielnie - jak ktoś będzie miał podobny proszę o kontakt chętnie pomogę.

**Reply by eklime** [reply]
To tak nie działa - rozpocząłeś wątek, to proszę skończ go należycie.
Napisz w trzech zdaniach co było problemem i jak go rozwiązałeś.
Tu dzielimy się wiedzą, a nie "pisz na priv"

**Reply by TheRealMamuth** [reply]
Hej - zamierzam - tylko wieczorem - dałem znać jakby ktoś się pochylał lub chciał coś pisać.

**Comment by TheRealMamuth** [comment, marked as answer]
Wyłączenie SPI w ustawieniach firewall pomogło, u mnie to kwestia chyba była po prostu takiego routera i oprogramowania. Router RV325  Gigabit Dual WAN VPN Router. Reguły w samym firewall nic nie dały - a wymieniona opcja była tylko jako on/off.

**Jeżeli chodzi o sama konfigurację w azure:**

1. VM z OpenVPN SA
2. Na interfejsie w Azure włączyć opcje forwarding
3. Dla subnetu dodać routingw Azure (o twojej sieci/sieciach) i jako brama wybrać wskazanie IP hosta w azure - wtedy będziemy sięgać wszystko co w Azure. Jak potrzeba do innej sieci to peering.
4. Nie jest to napisane ale możemy ustawić trasy routing na kliencie i w advance w ustawieniach OpenVPN SA w GUI (wtedy mamy zduplikowane wpisy w iptables) i może to powodować problemy.
5. NSG - dla obu maszyn dla sieci 192.168.2.0/24 w ogóle nie ruszane.

**Reply by eklime** [reply]
Z przyjemnych implementacji OpenVPN w Azure jest jeszcze VEEAMPN od Veeam. Śmiga w zasadzie od strzała ;)

**Reply by MariuszFerdyn** [reply]
Najlatwiejszy i najszybszy MultiVPN lacznie z OpenVPN... ale jak zobaczysz jak dziala L2TP od strzala, ktory jest wszedzie Out Of The Box nie chcesz juz OpenVPN.
